packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -112,6 +112,10 @@ export class PluginDeployerImpl implements PluginDeployer {
     }
 
     public async deploy(pluginEntry: string): Promise<void> {
+        if (this.pluginDeployerEntries.some(value => value.id() === pluginEntry)) {
```

Comment:
I wonder also whether it is a good place, VS Code extension can be loaded from different sources, the real ID is in package.json, i.e. publisher + name.